internal/model/log.go

```diff
@@ -34,10 +34,39 @@ var (
 type Log struct {
 	// Level holds the log level of the log event.
 	Level string
+	// Logger holds the name of the logger instance.
+	Logger string
+	Origin *LogOrigin
```

Comment:
Should these be non-pointer types, i.e. embedded structs? Using pointer types will likely force heap allocations, and it looks like we can rely on the zero values for omission.

Reply:
ACK on the non-pointer type. I thought about this and went with the current implementations for optional values thinking that the size of the struct won't make it escape to the heap.

>  i.e. embedded structs

Do you mean a move from nested to embedded structs? I personally find embedded structs introduce some unwanted complexity and I am not seeing a lot of benefits here (not aware of any performance difference either).

Reply:
> Do you mean a move from nested to embedded structs? I personally find embedded structs introduce some unwanted complexity and I am not seeing a lot of benefits here (not aware of any performance difference either).

I'm not sure why I said embedded now. Just a plain old struct field is what I actually had in mind.